statsdbinterface/rankings.py

```diff
@@ -114,50 +114,40 @@ def players_by_games(days):
 
 @cached(60)
 def modes_by_games(days):
-    re = redeclipse.versions.default
     first_game = first_game_in_days(days)
-    modes = extmodels.Mode.mode_list()
+    modes = models.Mode.query \
+        .filter(models.Mode.name.notin_(['edit', 'demo'])) \
+        .all()
     ret = []
     for mode in modes:
-        m = {
-            'name': mode,
-            'longname': re.modestr[re.modes[mode]],
-            'games': 0,
-            }
-        m['games'] = models.Game.query.filter(
-            db.func.re_mode(models.Game.id, mode),
-            models.Game.id >= first_game
-            ).count()
+        m = {'name': mode.name,
+             'longname': mode.longname,
+             'games': mode.games.filter(models.Game.id >= first_game)
+                                .count()}
         ret.append(m)
     return sorted(ret, key=lambda m: m['games'], reverse=True)
```

Comment:
So you say the cached `sorted()` object is copied when it's requested from the cache?

Still, memory wise the `sorted()` won't allow the GC to collect the `mode` instance until the cache timeout is reached.